routers/web/repo/pull.go

```diff
@@ -1024,10 +1024,14 @@ func CompareAndPullRequestPost(ctx *context.Context) {
 	)
 
 	headUser, headRepo, headGitRepo, prInfo, baseBranch, headBranch := ParseCompareInfo(ctx)
+	defer func() {
```

Comment:
This was safe previously but belt and braces seems reasonable

Reply:
The problem is if `ctx.Written()`, then headGitRepo is not closed.

Reply:
Iirc from my recent reviews of that code the head repo is not returned in that case.

But this is better and safer.